docs/architecture.md

```diff
@@ -0,0 +1,35 @@
+# OpenShift Hive Architecture
+
+OpenShift Hive is a Kubernetes operator built to provide a declarative API to provision, configure, reshape, and deprovision OpenShift clusters at scale. Hive's API is a set of Kubernetes Custom Resource Definitions, served by the Kubernetes apiserver.
+
+Hive leverages the [OpenShift Installer](https://github.com/openshift/installer) to perform cluster provisioning and deprovisioning.
+
+
+## Components
+
+ 1. **Hive Operator**: The first Hive pod to run is the hive-operator which is essentially a deployer. It manages the Kubernetes Deployments for other components to ensure they are properly configured and running. It also handles migrations and cleanup and anything else that might require programatic logic in a long term Hive deployment. Hive's operator reconciles on a HiveConfig CRD which provides some configuration options for Hive itself.
+ 1. **Hive Admission**: Hive Admission is a small stateless HTTP server used for CRD webhook validation. It's just is just to approve or deny creation or updates to our core CRDs.
+ 1. **Hive Controllers**: The core Hive controllers which reconcile all CRDs.
+
+## How Hive Works
+
+![Hive Architecture](hive-architecture.png "Hive Architecture")
+
+API callers create ClusterDeployments which contain the desired state for the cluster they would like to provision. Hive was intended to work in a multi-tenant environment, so ClusterDeployments and their associated secrets can be created in separate namespaces. This allows for protecting access to data with standard Kubernetes RBAC.
```

Comment:
s/Hive was intended to work in a multi-tenant environment/Hive is designed to work in a multi-tenant environment`